app/assets/javascripts/services/user_project_role_factory.js

```diff
@@ -0,0 +1,55 @@
+samson.factory('userProjectRoleFactory', function() {
+
+  function UserProjectRole(id, user_id, project_id, role_id) {
+    this.id = id;
+    this.user_id = user_id;
+    this.project_id = project_id;
+    this.role_id = role_id;
+  }
+
+  UserProjectRole.prototype.withUserName = function(user_name){
+    this.user_name = user_name;
+    return this;
+  };
+
+  UserProjectRole.prototype.withProjectName = function(project_name) {
+    this.project_name = project_name;
+    return this;
+  };
+
+  UserProjectRole.prototype.exists = function() {
+    //If id contains a valid number, this role has already been created in the database
+    return _.isNumber(this.id);
+  };
+
+  UserProjectRole.prototype.buildCreatePayload = function() {
+    return JSON.stringify(this, ['user_id', 'project_id', 'role_id']);
+  };
+
+  UserProjectRole.prototype.buildUpdatePayload = function() {
+    return JSON.stringify(this, ['role_id']);
+  };
```

Comment:
I don't think these 2 functions belong here, as the projectRolesService above is the one who knows what parameters it needs.